app/helpers/automated_tests_helper.rb

```diff
@@ -738,9 +737,53 @@ def self.launch_test(assignment, repo_path, call_on)
     end
   end
 
-  def self.process_result(result, grouping_id, assignment_id)
-  end
+  def self.process_result(raw_result)
+    result = Hash.from_xml(raw_result)
+    repo = @grouping.group.repo
+    @revision = repo.get_latest_revision
+    @revision_number = @revision.revision_number
+    raw_test_scripts = result['testrun']['test_script']
+
+    # Hash.from_xml will yield a hash if only one test script
+    # and an array therwise
+    if raw_test_scripts.nil?
+      return
+    elsif raw_test_scripts.is_a?(Array)
+      test_scripts = raw_test_scripts
+    else
+      test_scripts = [raw_test_scripts]
+    end
 
+    # For now, we just use the first test script for the association
+    test_result = TestResult.new
```

Comment:
This is good, but can you pass the attributes directly to `new`? E.g. `TestResult.new(grouping_id: @grouping.id, ...)`